upup/pkg/fi/assettasks/copyfile.go

```diff
@@ -197,26 +197,31 @@ func buildVFSPath(target string) (string, error) {
 		return target, nil
 	}
 
-	u, err := url.Parse(target)
-	if err != nil {
-		return "", fmt.Errorf("unable to parse url: %q", target)
-	}
-
 	var vfsPath string
 
-	// These matches only cover a subset of the URLs that you can use, but I am uncertain how to cover more of the possible
-	// options.
-	// This code parses the HOST and determines to use s3 or gs.
-	// URLs.  For instance you can have the bucket name in the s3 url hostname.
-	// We are translating known https urls such as https://s3.amazonaws.com/example-kops to vfs path like
-	// s3://example-kops
-	if u.Host == "s3.amazonaws.com" {
-		vfsPath = "s3:/" + u.Path
-	} else if u.Host == "storage.googleapis.com" {
-		vfsPath = "gs:/" + u.Path
+	// Matches all S3 regional naming conventions:
+	// https://docs.aws.amazon.com/general/latest/gr/rande.html#s3_region
+	// and converts to a s3://<bucket>/<path> vfsPath
+	s3VfsPath, err := vfs.VFSPath(target)
+	if err == nil {
+		vfsPath = s3VfsPath
 	} else {
-		glog.Errorf("unable to determine vfs path s3, google storage, and file paths are supported")
-		glog.Errorf("URLs starting with https://s3.amazonaws.com and http://storage.googleapis.com are transformed into s3 and gs URLs")
+		// These matches only cover a subset of the URLs that you can use, but I am uncertain how to cover more of the possible
+		// options.
+		// This code parses the HOST and determines gs URLs.
+		// For instance you can have the bucket name in the gs url hostname.
+		u, err := url.Parse(target)
+		if err != nil {
+			return "", fmt.Errorf("unable to parse url: %q", target)
+		}
+		if u.Host == "storage.googleapis.com" {
+			vfsPath = "gs:/" + u.Path
+		}
+	}
+
+	if vfsPath == "" {
+		glog.Errorf("unable to determine vfs path s3, google storage, and file paths are supported: %s", target)
+		glog.Errorf("URLs that adhere to S3 regional naming conventions (https://docs.aws.amazon.com/general/latest/gr/rande.html#s3_region) and http://storage.googleapis.com are transformed into s3 and gs URLs")
```

Comment:
The code looks awesome overall. Appreciate the tests. Can reword the error message as it does not tell the user here is the url u provided, here is what we expected. Also give more guidance on a proper gs URL.

Thanks

Awesome work!